app/models/work.rb

```diff
@@ -366,7 +366,7 @@ def log_changes(resource_compare, current_user)
 
   def log_file_changes(changes, current_user)
     return if changes.count == 0
-    WorkActivity.add_system_activity(id, changes.to_json, current_user.id, activity_type: "FILE-CHANGES")
+    WorkActivity.add_system_activity(id, changes.to_json, current_user&.id, activity_type: "FILE-CHANGES")
```

Comment:
@hectorcorrea I wonder if the parameter should really be `current_user_id` and the two places calling log_file_changes should pass the id.  The impression that a current_user will be sent in and then not could get confusing when maintaining the code.

Reply:
Good idea. I've made the change.